src/link/source.rs

```diff
@@ -0,0 +1,140 @@
+use std::collections::HashMap;
+use std::path::Path;
+
+use super::biblio::BiblioEntry;
+use super::reference::Reference;
+use crate::util::reader;
+
+#[derive(Debug, Default)]
+pub struct ReferenceSource {
+    base_path: String,
+    // text => reference
+    references: HashMap<String, Reference>,
+}
+
+impl ReferenceSource {
+    pub fn new(base_path: &str) -> Self {
+        ReferenceSource {
+            base_path: base_path.to_owned(),
+            ..Default::default()
+        }
+    }
+
+    fn load(&mut self, group: &str) {
+        let data_path = Path::new(&self.base_path).join(format!("anchors-{}.data", group));
+        let mut lines = reader::read_lines(data_path).unwrap();
+
+        while let Some(key) = lines.next() {
+            let key = key.unwrap();
+            let link_type = lines.next().unwrap().unwrap();
+            let spec = lines.next().unwrap().unwrap();
```

Comment:
This looks... very hacky. Can you comment on what format are you processing here?

Reply:
It seems like it's a bikeshed-specific data format, which is a little bit confusing. I will add more details when I figure it out.